test-engineering/contract-tests/client/tests/test_contile.py

```diff
@@ -3,34 +3,65 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
-from typing import List
+from typing import Callable, Dict, List
 
 import pytest
 import requests
-from models import Step
+from requests import Response as RequestsResponse
 
+from models import PartnerRecordsNotClearedError, Step
 
-@pytest.fixture(name="contile_host")
-def fixture_contile_host(request):
-    """Read the contile host from the pytest config."""
 
-    return request.config.option.contile_url
+@pytest.fixture(name="hosts", scope="session")
+def fixture_hosts(request) -> Dict[str, str]:
+    """Return a dict mapping from a service name to a host name."""
 
+    return {
+        "contile": request.config.option.contile_url,
+        "partner": request.config.option.partner_url,
+    }
 
-def test_contile(contile_host: str, steps: List[Step]):
+
+@pytest.fixture(name="clear_partner_records")
+def fixture_clear_partner_records(hosts: Dict[str, str]) -> Callable:
+    """Clear Contile request history on partner."""
+
+    partner_host = hosts["partner"]
+
+    def clear_partner_records():
+        r: RequestsResponse = requests.delete(f"{partner_host}/records/")
```

Comment:
Interesting. I'd expect flake8 to complain about the use of 1 character variable names. I think it used to do that. 😁

Reply:
Change Applied: [32becfb](https://github.com/mozilla-services/contile/pull/422/commits/32becfbca17c3055e4e064634dfe89aab78020e3)

I used the single character to stay consistent with the existing naming found in the `test_contile` method, but would actually prefer the full name `request`. This comment makes me think I can take the liberty of changing it throughout.